### PR TITLE
Josh/tamagui infrastructure

### PR DESCRIPTION
**Pull request by DynamisDevelopment**
Had to change file structure a bit, you'll probably have to update your deployment service. web should be able to function independently right now, so you can point to that and keep things simple for now.

When attempting to use styled from tamagui, browser returns error about useInsertionEffect. Currently working on a fix for that.

### CODE REVIEW

**Comment by kyh** [comment]
Thanks for this, couple thoughts:
1) Is the migration from `npm` -> `yarn` necessary?
2) Lets remove the .prettierc file so that we continue to use the default prettier configuration
3) Looks like you're commiting some `dist` files which we don't want

**Comment by DynamisDevelopment** [comment]
- I attempted to remove using yarn workspaces, but that turned out to be difficult. I can give it another shot.
- Removed prettier.rc
- Ah, you're right, thanks for noticing. Added to gitingore.

File: packages/app/.gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+node_modules/
```
Comment (kyh) [inline packages/app/.gitignore:1]: probably don't need this file, we can move these ignores to the root

**Comment by kyh** [comment]
The `.tamagui` folder also looks like it should be ignored?

**Comment by DynamisDevelopment** [comment]
> The `.tamagui` folder also looks like it should be ignored?

It's auto generated so yeah.

**Comment by DynamisDevelopment** [comment]
Took a bit of poking around and breaking stuff, since there's very little on this bug and the docs weren't much help, but I tracked down the useInsertionEffect error down to the tRPC instance, specifically the SSR. Had to disable it while I move forward with the migration, or until we can find a smarter fix for it. 

Apparently Tamagui uses useInsertionEffect in their compiled code and can't be removed.

I'm not too familiar with tRPC so if you have any ideas for a better solution that would be great!

**Comment by kyh** [comment]
I see, yeah we can disable ssr for now

**Comment by DynamisDevelopment** [comment]
> They've just recently updated their starters, as an FYI. not sure if this makes migrations a bit easier https://github.com/tamagui/tamagui/releases/tag/v1.1.5

Nice, I'll pull it over.

**Pull request closed without merging**